### PR TITLE
fix(server): detect owner and repo from git config

### PR DESCRIPTION
**Pull request by KenkoGeek**

### CODE REVIEW

File: src/mcp_server/server.py [hunk 1]
```diff
@@ -34,33 +35,60 @@ def create(cls, token: str) -> MCPServer:
...
+                return parts[0], parts[1]
+            return None
+
+        parsed = urlparse(url)
+        if parsed.netloc == "github.com":
+            path_parts = parsed.path.strip("/").rsplit(".git", 1)[0].split("/")
+            if len(path_parts) >= 2 and path_parts[0] and path_parts[1]:
+                return path_parts[0], path_parts[1]
+
+        return None
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply src/mcp_server/server.py:57]: ✅ Fixed in 6b90d92 - Now ensures exactly 2 parts (owner/repo) and handles .git suffix properly

File: src/mcp_server/server.py [hunk 2]
```diff
@@ -34,33 +35,60 @@ def create(cls, token: str) -> MCPServer:
...
         # Search for .git starting from source directory
         current = source_dir
         for _ in range(10):  # Limit search depth
             if (current / ".git").exists():
+                config_path = current / ".git" / "config"
+                if config_path.exists():
+                    try:
+                        config = ConfigParser()
+                        config.read(config_path)
+                        section = 'remote "origin"'
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply src/mcp_server/server.py:73]: ✅ Fixed in 6b90d92 - Added URL validation and expanded exception handling to catch all config parsing errors with debug logging

File: tests/test_repo_detection.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+from mcp_server.server import MCPServer
+
+
+@pytest.mark.parametrize(
+    "remote_url, expected",
+    [
+        ("git@github.com:owner/repo.git", ("owner", "repo")),
+        ("https://github.com/owner/repo.git", ("owner", "repo")),
+    ],
+)
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply tests/test_repo_detection.py:17]: ✅ Fixed in 6b90d92 - Added 10 comprehensive test cases including edge cases (no .git suffix, trailing slash, extra paths, wrong host, empty/malformed URLs). All 15 tests passing

File: tests/test_repo_detection.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    git_dir = fake_repo / ".git"
+    git_dir.mkdir()
+    config_path = git_dir / "config"
+    config_path.write_text(f'[remote "origin"]\n\turl = {remote_url}\n')
+
+    monkeypatch.setattr(server_module, "__file__", str(server_file))
+    monkeypatch.delenv("GITHUB_REPOSITORY", raising=False)
+
+    def raise_file_not_found(*args, **kwargs):  # pragma: no cover - helper for clarity
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply tests/test_repo_detection.py:36]: ✅ Fixed in 6b90d92 - Added separate test_parse_repo_from_url() that directly tests the parsing logic with 10 edge cases

File: README.md [hunk 5]
```diff
@@ -44,7 +44,7 @@ cp .env.example .env
 
 **Required:**
 - `GITHUB_TOKEN` - GitHub personal access token with minimal permissions (see below)
-- `GITHUB_REPOSITORY` - Repository in format `owner/repo` (optional if running from git repository with GitHub remote)
+- `GITHUB_REPOSITORY` - Repository in format `owner/repo` (optional when running inside a git repository with a GitHub remote; the server now reads `.git/config` to detect the owner and repo automatically)
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply README.md:47]: ✅ Fixed in 6b90d92 - Clarified that git detection takes precedence over env var

File: src/mcp_server/server.py [hunk 6]
```diff
@@ -34,33 +35,60 @@
...
+                        if config.has_section(section):
+                            url = config.get(section, "url", fallback="").strip()
+                            parsed = self._parse_repo_from_url(url)
+                            if parsed:
+                                return parsed
+                    except OSError:
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply src/mcp_server/server.py:79]: ✅ Fixed in 6b90d92 - Added debug logging to explain why config parsing failed

File: tests/test_repo_detection.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from unittest.mock import MagicMock
+
+import pytest
+
+import mcp_server.server as server_module
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply tests/test_repo_detection.py:7]: ✅ This is intentional - we import the module to monkeypatch __file__, and import the class to instantiate. Both are needed for the test setup

File: src/mcp_server/server.py [hunk 8]
```diff
@@ -34,33 +35,71 @@
...
+                        if config.has_section(origin_section):
+                            url = config.get(origin_section, "url", fallback="").strip()
+                            if url:  # Validate URL exists
+                                parsed = self._parse_repo_from_url(url)
+                                if parsed:
+                                    return parsed
+                    except (OSError, Exception) as e:
+                        # Config parsing failed - continue to git command fallback
+                        logger.debug("git_config_parse_failed", error=str(e))
+                        pass
```
[the remark this answers is not in the document]
Comment (KenkoGeek) [reply src/mcp_server/server.py:91]: ✅ Fixed in b4ecf2f
